smithy-build/src/main/java/software/amazon/smithy/build/transforms/FlattenNamespaces.java

```diff
@@ -164,9 +164,18 @@ private ShapeId updateNamespace(ShapeId shapeId, String namespace) {
         return shapeWalker.walkShapes(service).stream()
                 .filter(FunctionalUtils.not(Prelude::isPreludeShape))
                 .map(shape -> Pair.of(shape.getId(), updateNamespace(shape.getId(), config.getNamespace())))
+                .map(pair -> applyServiceRenames(pair, service))
```

Comment:
Nit: ideally we can destructure pairs before calling methods with them. That would give the left and right shape ID a name so it's easier to understand the code.